rsmtool/rsmpredict.py

```diff
@@ -25,6 +28,151 @@
+        RSMTool experiment.
+    df_feature_info : pandas DataFrame
+        A pandas dataframe containing the information regarding the model features.
+        The index of the dataframe should be the names of the features and
+        the columns should be:
+
+        - "sign" : 1 or -1.  Indicates whether the feature value needs to
+          be multiplied by -1.
+        - "transform" : :ref:`transformation <select_transformations_rsmtool>`
+          that needs to be applied to this feature.
+        - "train_mean", "train_sd" : mean and standard deviation for outlier
+          truncation.
+        - "train_transformed_mean", "train_transformed_sd" : mean and standard
+          deviation for computing z-scores.
+
+        This dataframe should be read from the "feature.csv" file under the
+        "output" directory of the previously run RSMTool experiment.
+    df_postprocessing_params : pandas DataFrame
+        A pandas dataframe containing the parameters needed for
+        post-processing the raw prediction. This dataframe should be read from the
+        "postprocessing_params.csv" file under the "output" directory of the
+        previously run RSMTool experiment.
+    predict_expected : bool, optional
+        Predict expected scores for classifiers that return probability
+        distributions over score. Note also that this assumes that the score
+        range consists of contiguous integers. Defaults to ``False``.
+    logger : logging object, optional
+        A logging object. If ``None`` is passed, get logger from ``__name__``.
+        Defaults to ``None``.
+
+    Returns
+    -------
+    dict[str, float]
+        A dictionary containing the raw, scaled, trimmed, and rounded
+        predictions for the input features. It contains the following
+        keys: "raw", "scale", "raw_trim", "scale_trim", "raw_trim_round",
+        and "scale_trim_round".
+
+    Raises
+    ------
+    ValueError
+        If ``input_features`` contains any non-numeric features.
+    """
+    # initialize a logger if none provided
+    logger = logger if logger else logging.getLogger(__name__)
+
+    # instantiate a feature preprocessor
+    preprocessor = FeaturePreprocessor(logger=logger)
+
+    # convert the given features to a data frame and add the "spkitemid" column
+    df_input_features = pd.DataFrame([input_features])
+    df_input_features["spkitemid"] = "RESPONSE"
+
+    # preprocess the input features so that they match what the model expects
+    try:
+        df_processed_features, _ = preprocessor.preprocess_new_data(
+            df_input_features, df_feature_info
+        )
+    except ValueError:
+        raise ValueError("Input features must not contain non-numeric values.") from None
+
+    # read the post-processing parameters
+    trim_min = df_postprocessing_params["trim_min"].values[0]
+    trim_max = df_postprocessing_params["trim_max"].values[0]
+    h1_mean = df_postprocessing_params["h1_mean"].values[0]
```

Comment:
is it possible to make passing these means and std devs optional in case we don't want/need scaled scores?

Reply:
Oh, yeah, good point!

Reply:
You always need the `trim_min` and `trim_max` since that's needed to trim the raw predictions. So, you will need to pass in the post processing data frame anyway.

Reply:
I was just thinking of a case where we can define the trim_min and trim_max outside of this data frame.. the couple instances I have seen of using RSMTool models in the engine have this information in the model config or defined in the service itself

Reply:
I see. Okay, how about this?

1. Make `trim_min`, `trim_max`, `trim_tolerance`, `train_predictions_mean`, train_predictions_sd`, `h1_mean` and `h1_sd` _all_ optional arguments for this function. All of them will be `None`, except for `trim_tolerance` which will have a default value of 0.4998. 
2. If you don't specify any of them, you just get the "raw" prediction.
3. If you specify `trim_min` and `trim_max`, you also get "raw_trim" and "raw_trim_round".
4. If you specify the other 4, you also get "scale", "scale_trim", and "scale_trim_round".

Reply:
that sounds great!